pkg/config/loader_test.go

```diff
@@ -0,0 +1,351 @@
+/*
+Copyright (c) 2018 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"io/ioutil"
+	"os"
+	"reflect"
+	"strings"
+	"testing"
+	"time"
+
+	autoheal "github.com/openshift/autoheal/pkg/apis/autoheal/v1alpha2"
+	batch "k8s.io/api/batch/v1"
+	core "k8s.io/api/core/v1"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestFiles(t *testing.T) {
+	l := NewLoader()
+
+	data0 := `
+      awx:
+        address: "http://test_address.com"`
+
+	data1 := `
+      awx:
+        proxy: "http://test-proxy.com:1234"`
+
+	temp_dir, _ := ioutil.TempDir(".", "test_dir")
```

Comment:
The convention for variable is camel case: `tempDir`.

Consider using `""` instead of `"."`, that way the temporary directory is created in `/tmp` instead of in the current working directory.